### PR TITLE
Add readstat and pyreadstat

### PR DESCRIPTION
**Pull request by swflint**
###### Description of changes

This adds the readstat and pyreadstat packages allowing users to read SAS, SPSS and Stata data files from within Python.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -10510,6 +10510,8 @@ with pkgs;
 
   read-edid = callPackage ../os-specific/linux/read-edid { };
 
+  readstat = callPackages ../applications/science/math/readstat { };
```
Comment (wegank) [inline pkgs/top-level/all-packages.nix:10513]:
```suggestion
  readstat = callPackage ../applications/science/math/readstat { };
```

File: pkgs/applications/science/math/readstat/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{ lib, stdenv, fetchFromGitHub, autoconf, automake, gettext, pkgconfig, libtool }:
+
+stdenv.mkDerivation rec {
+  name = "readstat";
+  version = "v1.1.8";
+
+  src = fetchFromGitHub {
+    owner = "WizardMac";
+    repo = "ReadStat";
+    rev = version;
```
Comment (SuperSandro2000) [inline pkgs/applications/science/math/readstat/default.nix:10]:
```suggestion
  version = "1.1.8";

  src = fetchFromGitHub {
    owner = "WizardMac";
    repo = "ReadStat";
    rev = "v${version}";
```

File: pkgs/applications/science/math/readstat/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    sha256 = "1r04lq45h1yn34v1mgfiqjfzyaqv4axqlby0nkandamcsqyhc7y4";
+  };
+
+  nativeBuildInputs = [ pkgconfig autoconf automake gettext ];
```
Comment (SuperSandro2000) [inline pkgs/applications/science/math/readstat/default.nix:14]:
```suggestion
  nativeBuildInputs = [ pkg-config autoreconfHook ];
```
already included in the hook

File: pkgs/applications/science/math/readstat/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  '';
+
+  meta = {
+    homepage = "https://github.com/WizardMac/ReadStat";
+    description = "Command-line tool (+ C library) for converting SAS, Stata, and SPSS files";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ swflint ];
+
+    mainProgram = "readstat";
+
```
Comment (SuperSandro2000) [inline pkgs/applications/science/math/readstat/default.nix:29]:
```suggestion
```
That is the default

File: pkgs/applications/science/math/readstat/default.nix [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    sha256 = "1r04lq45h1yn34v1mgfiqjfzyaqv4axqlby0nkandamcsqyhc7y4";
+  };
+
+  nativeBuildInputs = [ pkgconfig autoconf automake gettext ];
+  buildInputs = [ libtool gettext ];
+
+  preConfigure = ''
+    patchShebangs autogen.sh
+    ./autogen.sh
+  '';
```
Comment (SuperSandro2000) [inline pkgs/applications/science/math/readstat/default.nix:20]:
```suggestion
```
We can just use autoreconfHook

File: pkgs/development/python-modules/pyreadstat/default.nix [hunk 6]
```diff
@@ -0,0 +1,33 @@
+{ lib, buildPythonPackage, fetchFromGitHub, cython, zlib, pandas, readstat }:
+
+buildPythonPackage rec {
+  pname = "pyreadstat";
+  version = "v1.1.9";
+
+  src = fetchFromGitHub {
+    owner = "Roche";
+    repo = "pyreadstat";
+    rev = version;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pyreadstat/default.nix:10]:
```suggestion
  version = "1.1.9";

  src = fetchFromGitHub {
    owner = "Roche";
    repo = "pyreadstat";
    rev = "v${version}";
```

File: pkgs/development/python-modules/pyreadstat/default.nix [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+  meta = {
+    homepage = "https://github.com/Roche/pyreadstat";
+    description = "Python package to read SAS, SPSS and Stata files into pandas data frames using the readstat C library";
+    license = lib.licenses.asl20;
+    maintainers = with lib.maintainers; [ swflint ];
+
+    platforms = lib.platforms.unix;
+  };
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pyreadstat/default.nix:32]:
```suggestion
  };
```
that is the default

File: pkgs/development/python-modules/pyreadstat/default.nix [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+  propagatedBuildInputs = [
+    readstat
+    pandas
+  ];
+
+  buildInputs = [
+    cython
+    zlib
+  ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pyreadstat/default.nix:22]:
```suggestion
  nativeBuildInputs = [
    cython
  ];

  buildInputs = [
    zlib
  ];

  propagatedBuildInputs = [
    readstat
    pandas
  ];
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by swflint** [comment]
@SuperSandro2000 Thanks for the feedback.  I've gone ahead and made the changes.

File: pkgs/applications/science/math/readstat/default.nix [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+
+  src = fetchFromGitHub {
+    owner = "WizardMac";
+    repo = "ReadStat";
+    rev = "v${version}";
+    sha256 = "1r04lq45h1yn34v1mgfiqjfzyaqv4axqlby0nkandamcsqyhc7y4";
+  };
+
+  nativeBuildInputs = [ pkg-config autoreconfHook ];
+  buildInputs = [ libtool ];
```
Comment (SuperSandro2000) [inline pkgs/applications/science/math/readstat/default.nix:15]: You are sure about this? libtool normally goes to nativeBuildInputs and should be included in autoreconfHook.

Comment (swflint) [reply pkgs/applications/science/math/readstat/default.nix:15]: I barely write C/C++...  But that makes sense.  I'm assuming gettext goes in nativeBuildInputs as well then?

File: pkgs/applications/science/math/readstat/default.nix [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+  version = "1.1.8";
+
+  src = fetchFromGitHub {
+    owner = "WizardMac";
+    repo = "ReadStat";
+    rev = "v${version}";
+    sha256 = "1r04lq45h1yn34v1mgfiqjfzyaqv4axqlby0nkandamcsqyhc7y4";
+  };
+
+  nativeBuildInputs = [ pkg-config autoreconfHook libtool gettext ];
```
Comment (SuperSandro2000) [inline pkgs/applications/science/math/readstat/default.nix:14]:
```suggestion
  nativeBuildInputs = [ pkg-config autoreconfHook ];
```
Those are already included in autoreconfHook

**Review by SuperSandro2000 — COMMENTED** [review]
Also can you drop the v prefix in the commits. After that the PR should be mergeable.

File: pkgs/applications/science/math/readstat/default.nix (hunk 10 above)
Comment (swflint) [reply pkgs/applications/science/math/readstat/default.nix]: Ah, thanks.

**Pull request merged**